### ISSUE TITLE
Issue with JWT authentication.

### ISSUE THREAD

**Issue by kalimajayant** [docusign/code-examples-php#22]
On using Must authenticate, with JWT method, 

I get this error,

Deprecated: Non-static method Example\Controllers\Auth\DocuSign::getDefaultScopes() should not be called statically in C:\xampp\htdocs\docusign\code-examples-php-master\src\Services\JWTService.php on line 67
jb65ebbd0-f472-41d0-b25f-d05d36437685
Fatal error: Uncaught DocuSign\eSign\Client\ApiException: Error while requesting server, received a non successful HTTP code [400] with response Body: O:8:"stdClass":1:{s:5:"error";s:15:"invalid_request";} in C:\xampp\htdocs\docusign\code-examples-php-master\vendor\docusign\esign-client\src\Client\ApiClient.php:344 Stack trace: #0 C:\xampp\htdocs\docusign\code-examples-php-master\vendor\docusign\esign-client\src\Client\ApiClient.php(597): DocuSign\eSign\Client\ApiClient->callApi('/oauth/token', 'POST', Array, 'assertion=eyJ0e...', Array, NULL, NULL, true) #1 C:\xampp\htdocs\docusign\code-examples-php-master\src\Services\JWTService.php(76): DocuSign\eSign\Client\ApiClient->requestJWTUserToken('03ff0533-b1e8-4...', 'jb65ebbd0-f472-...', '-----BEGIN RSA ...', 'signature imper...') #2 C:\xampp\htdocs\docusign\code-examples-php-master\src\Services\JWTService.php(44): Example\Services\JWTService->configureJwtAuthorizationFlowByKey() #3 C:\xampp\htdocs\docusign\code-examples-php-master\src\Services\RouterService.php(310): Examp in C:\xampp\htdocs\docusign\code-examples-php-master\vendor\docusign\esign-client\src\Client\ApiClient.php on line 344

And when I open the url, in new tab, I am already authenticated.

Is this the recommended behaviour.
https://www.loom.com/share/15e589d870d749428a201ebdf6e08863

Also, I would like to know, The Jwt will be performed only once, or everytime. What If I just want to create an API on my server to send Envelope by post fields,

**Comment by frankois941** [comment]
Hi,

The sample is not working at **ALL** when using JWT

For what I see in the code, it was not tested for JWT authentification.

François

**Comment by AaronWDS** [comment]
We've made some updates to the launcher since this was posted. Please git pull to see the latest changes, update the config, use `composer update` to point to the latest sdk and dependencies, then try it again.  Please open a new issue if you encounter anything else. if you've made extensive changes, you can `git stash` to backup the changes you made, then `git pull` to grab these updates. Finally, run `git stash apply` to put back in your changes and create a merge commit as necessary.
